src/EventSubscriber/QuerySubscriber.php

```diff
@@ -130,6 +130,10 @@ public function queryAlter(QueryPreExecuteEvent $event) {
         'facet' => $facet,
       ]);
       $query_type_plugin->execute();
+      // Add query tags to the query.
```

Comment:
Query tags? Also, if you are gonna put a comment, say why you are adding these tags.

I see in multiple places you are adding the cache tags of the Facet entity to things. What relevant facet cache tags do we need? Do we need to invalidate any caches whenever a facet gets changed (which happens only as a deployment action)? Or are there tags I don't know about?